src/test/scala/mesosphere/mesos/ResourceMatcherTest.scala

```diff
@@ -1179,6 +1180,46 @@ class ResourceMatcherTest extends UnitTest with Inside {
       resourceMatchResponse shouldBe a[ResourceMatchResponse.NoMatch]
       resourceMatchResponse.asInstanceOf[ResourceMatchResponse.NoMatch].reasons.head shouldEqual DeclinedScarceResources
     }
+
+    "correctly match offers in case of app specific override and no Persistent Volume involved" in {
+
+      val overrideCases = Table(
+        ("gpu_scheduling_behavior", "GPU_SCHEDULING_BEHAVIOR", "expected"),
+        ("unrestricted", Some("restricted"), "NoMatch"),
+        ("unrestricted", None, "Match"),
+        ("restricted", Some("unrestricted"), "Match"),
+        ("restricted", None, "NoMatch")
+      )
+
+      forAll(overrideCases) { (gpuSchedulingBehavior, overrideLabel, expected) =>
```

Comment:
Nit: I'd like to use these cases outside of the `.. should ... in` pattern so we create new cases. It's not a blocker though.